File: day_01/01.py
#!/usr/bin/env python3
from itertools import cycle, accumulate
from collections import Counter

# Runs in ~0.09s

freqs = [int(i) for i in open('input.txt')]
print(sum(freqs))

c = Counter()
for i in accumulate(cycle(freqs)):
    if c[i]:
        print(i)
        break
    c[i] = True

File: day_01/01.R
#!/usr/bin/env Rscript

freqs <-read.table("input.txt")$V1
print(sum(freqs))

first_digit <- freqs[1]
cs <- cumsum(freqs)
# Inefficient - it works in chunks of size freqs
# where it should sample one-by-one
repeat {
    dup <- anyDuplicated(cs)
    if (dup) {
        print(cs[dup])
        break
    }
    freqs[1] <- first_digit + cs[length(cs)]
    cs <- c(cs, cumsum(freqs))
}
